File: dehaze/video.py
import cv2
import numpy as np
from non_local_dehazing import nonlocaldehazing
from atmlight_dcp import estairlight
# import im2double


def im2double(im):
    info = np.iinfo(im.dtype)  # Get the data type of the input image
    # Divide all values by the largest possible value in the datatype
    return im.astype(np.float) / info.max


i = 0
# videoFReader = cv2.VideoCapture('vid.MP4')
# fourcc = cv2.VideoWriter_fourcc('M', 'J', 'P', 'G')
# v = cv2.VideoWriter('output.avi', fourcc, 12, (320, 240))
# # FrameRate = 12
# gamma = 1
# while(videoFReader.isOpened()):
#     print i
#     i = i + 1
#     if i == 30:
#         break
#     ret, image_hazy = videoFReader.read()
#     gamma = 1
#     if ret:
#         A = np.reshape(estairlight(np.power(im2double(image_hazy), gamma)), (1, 1, 3))
#         (image_dehazed, transmission_refined) = nonlocaldehazing(image_hazy, A, gamma)
#         v.write(image_dehazed)

# v.release()
# videoFReader.release()

# image
gamma = 1
image_hazy = cv2.imread('pumpkins.png')
A = np.reshape(estairlight(np.power(im2double(image_hazy), gamma)), (1, 1, 3))
(image_dehazed, transmission_refined) = nonlocaldehazing(image_hazy, A, gamma)

cv2.imshow('imagedehazed', image_dehazed)
cv2.waitKey(0)
cv2.destroyAllWindows()

File: dehaze/video.m
% videoFReader = vision.VideoFileReader('vid.MP4');
% videoPlayer = vision.VideoPlayer;
% v = VideoWriter('newfile.avi','Motion JPEG AVI');
% 
% i=0;
% v.FrameRate = 12;
% open(v);
% while ~isDone(videoFReader)
%    i=i+1;
%    image_hazy = step(videoFReader);
% %    I = rgb2gray(frame);
% %    display(i);
%    %step(videoPlayer,frame);
%    
%    gamma = 1;
%    A = reshape(estimate_airlight(im2double(image_hazy).^(gamma)),1,1,3);
% 
%    % Dehaze the image	
%    [image_dehazed, transmission_refined] = non_local_dehazing(image_hazy, A, gamma );
% 
%    % Display results
% %    figure('Position',[50,50, size(image_hazy,2)*3 , size(image_hazy,1)]);
% %    subplot(1,3,1); imshow(image_hazy);    title('Hazy input')
% %    subplot(1,3,2); imshow(image_dehazed); title('De-hazed output')
% %    subplot(1,3,3); imshow(transmission_refined); colormap('jet'); title('Transmission')
% 
%    
%    writeVideo(v,image_dehazed);
% end
% % imshow(frame);
% close(v);
% release(videoFReader);
% release(videoPlayer);

image_hazy = imread('img.png');

gamma = 1;
A = reshape(estimate_airlight(im2double(image_hazy).^(gamma)),1,1,3);

% Dehaze the image	
[image_dehazed, transmission_refined] = non_local_dehazing(image_hazy, A, gamma );

imshow(image_dehazed);